File: FinalProject-LibraryManagementSystem/Spring/reservation-service/src/main/java/com/devni/tlp/finalproject/reservationservice/shared_model/Book.java
package com.devni.tlp.finalproject.reservationservice.shared_model;

import lombok.Data;

import java.util.List;

@Data
public class Book {
    private Integer id;
    private String title;
    private List<Author> authors;
    private String ISBN;
    private static int noOfCopies;
//    private int noOfCopies;
    private User user;
    private boolean isDeleted;

    public static int getNoOfCopies() {
        return noOfCopies;
    }

    public static void setNoOfCopies(int noOfCopies) {
        Book.noOfCopies = noOfCopies;
    }
}

File: FinalProject-LibraryManagementSystem/Angular/library-mgt-system/src/app/models/Book.ts
export class Book {
  id: number;
  title: string;
  authors: Authors[];
  isbn: string;
  noOfCopies: number;
  isDeleted: boolean;
  picByte: string;
  retrievedImage: string;
}

export interface Authors {
  authorName: string;
}
